### PR TITLE
feat(gatsby): Add meta tag for development

### PR DESCRIPTION
**Pull request by Darking360**
# Feature: Add in ability to signal development environment 💡

## Description

It adds a `metatag` into the development headers so we can identify whether the app was generated non-optimized or is using development mode.

## Related Issues

This is an issue created by @DSchau in a request by @kitze that you can find [here](https://github.com/gatsbyjs/gatsby/issues/16137) and more information in his original [tweet](https://twitter.com/thekitze/status/1154733782815117317).

## Work done

- Add metatag `name` to specify the current environment for the app.
- Updated snapshots.
- Add tests to check the feature, these were added in [static entry](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/__tests__/static-entry.js) tests inside the `cache-dir` directory.

### Using gatsby-dev

Using `gatsby-dev` with a `gatsby-universal` starter, I managed to get the feature working as expected, attaching an image so you can check.

gatsby metatag environment

While on the other hand, generating a production build and uploading it to now.sh, it does not contain that metatag, for this I filtered the metatags with:

```
Array.from(document.getElementsByTagName('meta')).filter(({ name }) => name === 'note')
```

gatsby production metatag

### CODE REVIEW

File: packages/gatsby/cache-dir/develop-static-entry.js [hunk 1]
```diff
@@ -27,7 +27,9 @@ try {
 Html = Html && Html.__esModule ? Html.default : Html
 
 export default (pagePath, callback) => {
-  let headComponents = []
+  let headComponents = [
+    <meta key="env:development" name="note" content="env:development" />,
```
Comment (sidharthachatterjee) [inline packages/gatsby/cache-dir/develop-static-entry.js]: Can we stick to `<meta name="note" content="environment=development">` as proposed in the issue?

Comment (Darking360) [reply packages/gatsby/cache-dir/develop-static-entry.js]: Oh! My bad 😅 Sure sure, let me update it and let you know ✍️

**Comment by Darking360** [comment]
@sidharthachatterjee PR updated. Sorry for the confusion 😅

I changed the key for the metatag to `environment-note` so it makes more sense 😁

**Comment by sidharthachatterjee** [comment]
@Darking360 Should we set the key to “environment”? Not sure if the “-note” makes sense out of context. Sorry for being difficult on this! 😅

**Comment by sidharthachatterjee** [comment]
Published in `gatsby@2.13.70`

**Comment by crosumo** [comment]
Hi! When i see the source code i cant find metatags.
But i can see metatags when i inspect element.
Its in localhost. Its normal? thx^^

**Pull request merged**